### PR TITLE
Add options to controll the database connections (open,idle,timeout)

### PR DESCRIPTION
**Pull request by 6543**
this allow more fine control and make sure woodpecker do not crash in high use scenarios where the engine try to create more connection as the database do allow ... (mariadb=100)

### CODE REVIEW

**Comment by 6543** [comment]
the defaults i took form gitea

File: cmd/server/flags.go [hunk 1]
```diff
@@ -36,6 +36,24 @@ var flags = append([]cli.Flag{
 		Name:    "log-xorm-sql",
 		Usage:   "enable xorm sql command logging",
 	},
+	&cli.IntFlag{
+		Sources: cli.EnvVars("WOODPECKER_XORM_MAX_CONNECTIONS"),
```
Comment (anbraten) [inline cmd/server/flags.go:40]: Should get expert prefix and description hint.

Comment (6543) [reply cmd/server/flags.go:40]: well I dont thing so ... if you config more connections for your db you can increase it here ...

and that's just an normal admin task

Comment (6543) [reply cmd/server/flags.go]: @qwerty287 @zc-devs done

**Pull request merged**